api/src/services/ethNetworkService.js

```diff
@@ -1,16 +1,71 @@
 const Web3 = require('web3');
-const { ETHEREUM_HTTP_PROVIDER_URL, ETHEREUM_CERE_TOKEN_CONTRACT_ADDRESS } = process.env;
-const web3 = new Web3(new Web3.providers.HttpProvider(ETHEREUM_HTTP_PROVIDER_URL));
-const erc20TokenAbi = require('./../contracts/erc20TokenAbi.json');
-const { decimal } = require("./../constants/config");
-const contract = new web3.eth.Contract(
-  erc20TokenAbi,
-  ETHEREUM_CERE_TOKEN_CONTRACT_ADDRESS
-);
+const { blockchains } = require("../config");
+const erc20TokenAbi = require("../config/contracts/erc20TokenAbi.json");
+const { decimals, blockchainNames } = require("../config/blockchains");
+const networkParams = new Map();
+
+async function init() {
+  const promises = [];
+  const ethereum = blockchains[1];
+  const ethBlockchains = [ethereum];
+  ethBlockchains.forEach(ethBlockchain => {
+    ethBlockchain.networks.forEach(network => {
+      const { rpcUrl, cereTokenContractAddress} = network;
+      const promise = async() => ({
+        blockchain: ethBlockchain.name,
+        name: network.name,
+        rpc: await initNetwork(rpcUrl, cereTokenContractAddress)
+      });
+      promises.push(promise());
+    });
+  });
+  const res = await Promise.all(promises);  
+  res.forEach(network => {
+    const providerId = getProviderId(network.blockchain, network.name);
+    networkParams.set(providerId, network.rpc);
+  });  
+}
+
+async function initNetwork(url, cereTokenContractAddress) {
+  return initProvider(url, cereTokenContractAddress);
+}
+
+async function initProvider(url, cereTokenContractAddress) {
+  const provider = new Web3.providers.HttpProvider(url);
+  const web3 = new Web3(provider);
+  const contract = new web3.eth.Contract(
+    erc20TokenAbi,
+    cereTokenContractAddress
+  );
+  console.log(`Connected to ${url}`);
+  return { web3, contract };
+}
+
+function getProviderId(blockchain, network) {
+  return  `${blockchain}_${network}`;
+}
 
 module.exports = {
-  getCEREBalanceOf: async (address) => {
+  init,
+  getCEREBalanceOf: async (blockchain, network, address) => {
```

Comment:
Let's get rid of this method and use `getErc20Balance` instead.

Reply:
resolved